More aggressively market functorch.vmap when torch.vmap gets called

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#67347 More aggressively market functorch.vmap when torch.vmap gets called**

This PR:
- changes the warning when torch.vmap gets called to suggest using
functorch.vmap
- changes the warning when a batching rule isn't implemented to suggest
using functorch.vmap

Test Plan:
- test/test_vmap.py

Differential Revision: [D31966603](https://our.internmc.facebook.com/intern/diff/D31966603)

zou3519 committed:
More aggressively market functorch.vmap when torch.vmap gets called

This PR:
- changes the warning when torch.vmap gets called to suggest using
functorch.vmap
- changes the warning when a batching rule isn't implemented to suggest
using functorch.vmap

Test Plan:
- test/test_vmap.py

[ghstack-poisoned]

samdow commented on aten/src/ATen/BatchedFallback.cpp:
```diff
@@ -66,13 +66,16 @@ static bool isInplaceOp(const c10::FunctionSchema& schema) {
+  TORCH_WARN("There is a performance drop because we have not yet implemented ",
+             "the batching rule for ", schema.operator_name(), ". ",
+             "We've moved development of vmap to to functorch "
+             "(https://github.com/pytorch/functorch), please try functorch.vmap) "
```
nit: Unmatched paren after `functorch.vmap`

samdow commented on aten/src/ATen/BatchedFallback.cpp:
```diff
@@ -66,13 +66,16 @@ static bool isInplaceOp(const c10::FunctionSchema& schema) {
+             "We've moved development of vmap to to functorch "
+             "(https://github.com/pytorch/functorch), please try functorch.vmap) "
+             "instead and/or file ",
+             "us an issue on GitHub so that we can prioritize its implementation.");
```
nit: "file us an issue" -> "file an issue"

samdow commented on torch/_vmap_internals.py:
```diff
@@ -158,9 +158,10 @@ def vmap(func: Callable, in_dims: in_dims_t = 0, out_dims: out_dims_t = 0) -> Ca
+        and contains significant performance improvements.
         Please give that a try if that is what you're looking for.
 
         Furthermore, if you're interested in using vmap for your use case,
```
Do we want to update this to send people to functorch instead so things don't get lost in a pytorch issue?

zou3519 replied:
That pytorch issue directs folks back to functorch (and also has instructions for how to find us on Slack) so I think it's fine to leave it for now.

samdow reviewed (approved):
LGTM!

zou3519 committed:
Update on "More aggressively market functorch.vmap when torch.vmap gets called"


This PR:
- changes the warning when torch.vmap gets called to suggest using
functorch.vmap
- changes the warning when a batching rule isn't implemented to suggest
using functorch.vmap

Test Plan:
- test/test_vmap.py

Differential Revision: [D31966603](https://our.internmc.facebook.com/intern/diff/D31966603)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31966603).
